### PR TITLE
Use Catalog to define dependencies

### PR DESCRIPTION
**Pull request by gfran93**
new catalog dependencies

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ok2CurlInterceptor = "0.7.0"
+okhttp3 = "4.9.1"
+okIo = "2.10.0"
+once = "1.3.0"
+stetho = "1b2c025a14"
+timber = "4.7.1"
+
+[libraries]
+
+swissKnife-core = {module = "com.github.xmartlabs.AndroidSwissKnife:swissknife-core", version.ref = "androidSwissKnife"}
```
Comment (mirland) [inline gradle/libs.versions.toml:24]: what about adding spaces?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+okhttp3-stetho = {module = "com.github.mathroule.stetho:stetho-okhttp3", version.ref = "stetho"}
+
+[bundles]
+
+androidSwissKnife = ["swissKnife-core", "swissKnife-dataStore"]
+compose = ["compose-compiler", "compose-constraintLayout", "compose-foundation", "compose-material",
+  "compose-navigation", "compose-runtime", "compose-tooling", "compose-ui", "compose-viewmodel"]
+kotlin = ["kotlin-core", "kotlin-coroutines"]
+okhttp3 = ["okhttp3-core", "okhttp3-logginInterceptor", "okhttp3-ok2CurlInterceptor", "okhttp3-urlConnection",
+  "okhttp3-stetho"]
```
Comment (mirland) [inline gradle/libs.versions.toml:66]: I'd use two bundles, okHttp and okHttpDebug

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+utils = ["coil", "okIo", "once", "stetho"]
```
Comment (mirland) [inline gradle/libs.versions.toml:67]: okIo can be defined in the same bundle as retrofit and okhttp

File: versions.gradle [hunk 4]
```diff
@@ -12,7 +12,6 @@ ext.androidVersions = androidVersions
 
 ext.deps = [:]
```
Comment (mirland) [inline versions.gradle:13]: This file should be removed, I assume this is a WIP branch, am I right? Add WIP to the branch name

**Review by mirland — APPROVED** [review]
It looks great!

File: versions.gradle (hunk 4 above)
Comment (gfran93) [reply versions.gradle:13]: I suppose that this file will be removed after migrating all the dependencies

File: versions.gradle [hunk 5]
```diff
@@ -10,178 +10,17 @@ androidVersions.buildTools = '30.0.2'
 androidVersions.targetSdk = 30
 ext.androidVersions = androidVersions
 
-ext.deps = [:]
```
Comment (mirland) [inline versions.gradle:13]: I think we should remove this file, we should locate the androidVersions in the gradle.properties file or in the main build.gradle file. What do you think?

File: versions.gradle [hunk 6]
```diff
@@ -10,178 +10,17 @@ androidVersions.buildTools = '30.0.2'
...
-retrofit2.core = "com.squareup.retrofit2:retrofit:$versions.retrofit2"
-retrofit2.gsonAdapter = "com.squareup.retrofit2:converter-gson:$versions.retrofit2"
-deps.retrofit2 = retrofit2
-
-
-ext.deps = deps
-
 ext.buildscriptDeps = [:]
 buildscriptDeps.dependencyUpdates = "com.github.ben-manes:gradle-versions-plugin:$versions.dependencyUpdates"
 buildscriptDeps.detekt = "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:$versions.detekt"
```
Comment (mirland) [inline versions.gradle:26]: this can be declared in the catalog file, and then you can consume in the main build.gradle file

**Review by mirland — COMMENTED** [review]
Awesome, it looks pretty cool :sunglasses:

**Review by sante — APPROVED** [review]

**Review by f7deleon — APPROVED** [review]

**Pull request merged**